iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergSerDe.java

```diff
@@ -224,7 +232,7 @@ public Writable serialize(Object o, ObjectInspector objectInspector) {
     Deserializer deserializer = deserializers.get(objectInspector);
     if (deserializer == null) {
       deserializer = new Deserializer.Builder()
-          .schema(tableSchema)
+          .schema(isDelete ? deleteSchema : tableSchema)
```

Comment:
Why not use the `projectedSchema` here?

Reply:
`projectedSchema` is only local variable inside `initialize()` and not available here

Reply:
would it make sense to keep the `projectedSchema` attribute and remove the `isDelete` and the `deleteSchema`?